### ISSUE TITLE
[BUG] `PLAYWRIGHT_DOWNLOAD_CONNECTION_TIMEOUT` env variable is not a connection timeout but also download timeout

### ISSUE THREAD

**Issue by tomaspetrusek-ext43791** [microsoft/playwright#28619]
### System info
- Playwright Version: 1.40.1
- Operating System: macOS 13.5
- Browser: [All, Chromium, Firefox, WebKit]
- Other info:

### Source code

For `npx playwright install` the `PLAYWRIGHT_DOWNLOAD_CONNECTION_TIMEOUT` env variable is not a connection timeout but also a download timeout.

**Steps**
- PLAYWRIGHT_DOWNLOAD_CONNECTION_TIMEOUT=100 npx playwright install chromium
- see how it fails to install due to short time limit at ~8%... in case it would be successful, decrease the time.

**Expected**

if connection is established, the timeout should be ignored ... and wait for the 100MB download.

**Actual**

the connection is reset during download

**Possible solutions (suggested)**
- fix behaviour
- fix naming

**Comment by mxschmitt** [comment]
Could you give us more details on the scenario you are facing with this issue?

**Comment by tomaspetrusek-ext43791** [comment]
I had an (unrelated) issue (https://github.com/microsoft/playwright/issues/28620 , https://github.com/microsoft/playwright/issues/20863) and since the failure `Error [ERR_SOCKET_CLOSED]: Socket is closed` came extremely fast, I thought lets give it a bit more time to connect :-). Not sure how, but I came across `PLAYWRIGHT_DOWNLOAD_CONNECTION_TIMEOUT` and I thought - based on the name - that it would do exactly what I wanted... to try connecting for longer (I didn't know the problem was with a proxy setting). So I added this to my CLI (timeout 10) and continued trying all kinds of different stuff until it eventually started working - almost... and than I realized:
- the timeout is not in seconds, but in milliseconds (not expected on CLI, prefectly fine in code - but I agree it is very subjective)
- it cancels a started connection - so when I saw a progress-bar with 3% and it failed, I knew it is my fault and everything should work just fine. So again, I just didn't expect this behaviour based on the name of the env variable.

I haven't found any specs for this, maybe it is not even intended for public use, I just wanted to let you know ;-).

_Note: I guessed it could be used for some CI runners not to wait too long to establish a connection, but on the other hand, if the download is already in progress they probably want to wait for it. Anyway, both options could be useful, connection timeout and also download timeout... I would just suggest to improve the naming._
